api/v1alpha1/dataplane_types.go

```diff
@@ -20,7 +20,8 @@ type ValueFrom struct {
 // KubernetesClusterSpec defines the configuration for the target Kubernetes cluster
 type KubernetesClusterSpec struct {
 	// Server is the URL of the Kubernetes API server
-	Server string `json:"server"`
+	// +optional
+	Server *string `json:"server,omitempty"`
```

Comment:
With this change if the both control plane and data plane is deployed in a single cluster, kube api url is auto resolved. This would eliminate finding the current issue of setting the kube api url correctly from install/add-default-dataplane.sh